k9mail-library/src/main/java/com/fsck/k9/mail/internet/ListHeaders.java

```diff
@@ -0,0 +1,46 @@
+package com.fsck.k9.mail.internet;
+
+import android.net.MailTo;
+
+import com.fsck.k9.mail.Address;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Intended to cover:
+ *
+ * RFC 2369
+ * The Use of URLs as Meta-Syntax for Core Mail List Commands
+ * and their Transport through Message Header Fields
+ * https://www.ietf.org/rfc/rfc2369.txt
+ *
+ * This is the following fields:
+ *
+ * List-Help
+ * List-Subscribe
+ * List-Unsubscribe
+ * List-Post
+ * List-Owner
+ * List-Archive
+ *
+ * Currently only provides a utility method for List-Post
+ **/
+public class ListHeaders {
+    public static final String LIST_POST_HEADER = "List-Post";
+    private static final Pattern mailtoContainerPattern = Pattern.compile("<(.+)>");
+
+    public static Address[] parsePostAddress(String headerValue) {
+        if (headerValue == null || headerValue.isEmpty()) {
+            return new Address[]{};
+        } else {
```

Comment:
> As a general code style preference I prefer extracting functions to skipping on else statements.

:+1: 

> You have to read the code inside the if block to know whether it will be affected by the code below. See discussion here: http://programmers.stackexchange.com/a/255528

I agree that indiscriminately removing `else` statements is a bad idea. However, in this specific case we're using guard conditionals to exit the method early. I do believe that omitting `else` makes the code easier to read in that case. I think another answer to the question you linked to, sums that up nicely:
http://programmers.stackexchange.com/a/255532

For the second `if` statement you could argue whether or not omitting the `else` statement makes the code more readable. I think the following is fine, too.

```java
Matcher matcher = mailtoContainerPattern.matcher(headerValue);
if (matcher.find()) {
    String mailtoUri = matcher.group(1);
    return extractAddressesFromMailToUri(mailtoUri);
} else {
    return new Address[0];
}
```

In those cases short `if` and `else` bodies are important, though.

> (Generally for example, if I see a comment demarcating a block of code I look to pull out a function which uses that comment as it's name.)

:100: :+1: